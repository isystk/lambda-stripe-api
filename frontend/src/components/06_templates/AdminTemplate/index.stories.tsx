import { Meta, Story } from '@storybook/react'
import React from 'react'
import AdminTemplate from './index'
import MainService from '@/services/main'

export default {
  title: '06_templates/AdminTemplate',
  component: AdminTemplate,
} as Meta

const Template: Story = () => {
  const main = new MainService(() => ({}))
  const props: AdminTemplate = { main }
  return <AdminTemplate {...props} />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
