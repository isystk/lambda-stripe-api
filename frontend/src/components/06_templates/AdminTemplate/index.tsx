import Circles from '@/components/02_interactions/Circles'
import React, { FC } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { connect } from '@/components/hoc'
import HtmlSkeleton, {
  NoIndex,
  Title,
} from '@/components/05_layouts/HtmlSkeleton'
import MainService from '@/services/main'
import { adminMenuItems } from '@/constants/menu'
import Logo from '@/components/01_atoms/Logo'
import { Url } from '@/constants/url'
import { useI18n } from '@/components/i18n'
import { useRouter } from 'next/router'
import Breadcrumb, {
  type BreadcrumbItem,
} from '@/components/01_atoms/Breadcrumb'
import DropDown from '@/components/01_atoms/DropDown'

/** AdminTemplateProps Props */
export type AdminTemplateProps = WithChildren & {
  main: MainService
  title: string
  breadcrumb?: BreadcrumbItem[]
}
/** Presenter Props */
export type PresenterProps = AdminTemplateProps & {
  t
  logout
  router
}

/** Presenter Component */
const AdminTemplatePresenter: FC<PresenterProps> = ({
  children,
  main,
  t,
  title,
  breadcrumb = [],
  logout,
  router,
  ...props
}) => (
  <HtmlSkeleton>
    <Title>{`${title}－Admin`}</Title>
    <NoIndex />
    <div className="h-screen">
      <div className="h-16 bg-base p-4 md:p-0 flex">
        <Logo link={Url.AdminHome} />
        {main.user && (
          <div className="ml-auto p-3 md:p-6">
            <DropDown
              label={main.user.userName}
              items={[{ label: 'ログアウト', link: () => logout() }]}
            />
          </div>
        )}
      </div>
      <div className="grid grid-cols-12">
        <div className="col-span-2 bg-base hidden md:block">
          <div className="p-4 border-t">
            <ul className="list-none">
              {adminMenuItems.map(({ label, href, target }, idx) => (
                <li className="my-6" key={idx}>
                  <a
                    href="#"
                    target={target}
                    rel={target ? 'noreferrer' : ''}
                    className="break-words whitespace-pre-wrap text-gray-700 font-bold whitespace-nowrap"
                    onClick={(e) => {
                      e.preventDefault()
                      router.push(href)
                    }}
                  >
                    {t(label)}
                  </a>
                </li>
              ))}
            </ul>
          </div>
        </div>
        <div className="col-span-12 md:col-span-10">
          <div
            className="flex items-center justify-center"
            style={{ height: 'calc(100vh - 4rem)' }}
          >
            <Circles>
              {main.user && <Breadcrumb items={breadcrumb} />}
              <div className="py-8 md:p-8 w-full">{children}</div>
            </Circles>
          </div>
        </div>
      </div>
    </div>
  </HtmlSkeleton>
)

/** Container Component */
const AdminTemplateContainer: React.FC<
  ContainerProps<AdminTemplateProps, PresenterProps>
> = ({ presenter, main, children, ...props }) => {
  const router = useRouter()
  const { t } = useI18n('Admin')
  const logout = async () => {
    await main.logout()
    location.reload()
  }

  return presenter({
    children,
    main,
    t,
    logout,
    router,
    ...props,
  })
}

export default connect<AdminTemplateProps, PresenterProps>(
  'AdminTemplate',
  AdminTemplatePresenter,
  AdminTemplateContainer
)
