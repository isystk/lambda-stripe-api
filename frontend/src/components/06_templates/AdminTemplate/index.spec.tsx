import React from 'react'
import renderer from 'react-test-renderer'
import AdminTemplate, { AdminTemplateProps } from './index'

import MainService from '@/services/main'
import { Provider } from 'react-redux'
import { store } from '@/stores'

describe('AdminTemplate', () => {
  it('Match Snapshot', () => {
    const main = new MainService(() => ({}))
    const props: { main: MainService } = { main }
    const component = renderer.create(
      <Provider store={store}>
        <AdminTemplate {...props} />
      </Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
