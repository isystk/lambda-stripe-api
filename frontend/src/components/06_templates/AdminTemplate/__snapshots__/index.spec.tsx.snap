// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPageTemplate Match Snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-screen"
  >
    <div
      class="h-16 bg-base p-4 md:p-0 flex"
    >
      <a
        class="css-0"
        href="#"
      >
        <div
          class="w-20 md:w-40"
        >
          <img
            class="css-s08p0c "
            src="/images/logo.png"
          />
        </div>
      </a>
    </div>
    <div
      class="grid grid-cols-12"
    >
      <div
        class="col-span-2 bg-base hidden md:block"
      >
        <div
          class="p-4 border-t"
        >
          <ul
            class="list-none"
          >
            <li
              class="my-6"
            >
              <a
                class="break-words whitespace-pre-wrap text-gray-700 font-bold whitespace-nowrap"
                href="#"
                rel=""
              >
                Product List
              </a>
            </li>
            <li
              class="my-6"
            >
              <a
                class="break-words whitespace-pre-wrap text-gray-700 font-bold whitespace-nowrap"
                href="#"
                rel=""
              >
                Subscribe List
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="col-span-12 md:col-span-10"
      >
        <div
          class="flex items-center justify-center"
          style="height: calc(100vh - 4rem);"
        >
          <div
            class="css-1wwfi4"
          >
            <ul
              class="css-10ffb9g"
            >
              <li
                class="css-i6ammj"
              />
              <li
                class="css-l07tar"
              />
              <li
                class="css-1hre2z7"
              />
              <li
                class="css-p2xyi8"
              />
              <li
                class="css-u8edpl"
              />
              <li
                class="css-ik3prs"
              />
              <li
                class="css-1wh69e5"
              />
              <li
                class="css-1cg3at6"
              />
              <li
                class="css-1iaru34"
              />
              <li
                class="css-1io9drp"
              />
            </ul>
            <div
              class="relative"
            >
              <div
                class="py-8 md:p-8 w-full"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
